homeassistant/components/homekit_controller/connection.py

```diff
@@ -493,21 +492,16 @@ async def async_load_platform(self, platform):
     async def async_load_platforms(self):
         """Load any platforms needed by this HomeKit device."""
         tasks = []
-        for accessory in self.accessories:
-            for service in accessory["services"]:
-                try:
-                    stype = normalize_uuid(service["type"])
-                except KeyError:
-                    stype = service["type"].upper()
-
-                if stype in HOMEKIT_ACCESSORY_DISPATCH:
-                    platform = HOMEKIT_ACCESSORY_DISPATCH[stype]
+        for accessory in self.entity_map.accessories:
+            for service in accessory.services:
+                if service.type in HOMEKIT_ACCESSORY_DISPATCH:
+                    platform = HOMEKIT_ACCESSORY_DISPATCH[service.type]
                     if platform not in self.platforms:
                         tasks.append(self.async_load_platform(platform))
```

Comment:
We could consider collecting the new platform names to load instead of collecting tasks, and then create all tasks and gather them at the end when we have all platforms. Now we collect the tasks two times.

Reply:
That sounds good, I've made a note in my backlog to come back to it when i'm happy with my typing prep changes on the HA side.